main.cpp

```diff
@@ -173,36 +246,91 @@ struct VendingMachine
         int inventory = 5;
         float priceEuros = 2.5f;
         int itemNumber = 13;
+        bool isDisabled = false;
 
-        bool distributeItems(int numberOfItems); // returns true if item successfully distributed
+        bool distributeItems(int numberOfItems); 
 
-        int updateInventory(int itemAdded); // returns the new inventory
+        void stockUp(int itemsAdded); 
 
-        bool disable(std::string cause); // return true if the dispenser is disabled
+        void disable(std::string cause); 
     };
 
-    // 3 things it can do:
-    // charge customer for an item
-    float chargeCustomerEuros(int item); // returns the price for the item
+    float chargeCustomerEuros(ItemDispenser item, int numberOfItems); 
 
-    // dispense item
-    bool dispenseItem(ItemDispenser itemDispenser, int numberOfItems); // returns true if the items are sucessfully dispensed
+    bool dispenseItem(ItemDispenser itemDispenser, int numberOfItems); 
 
-    // refrigerate at a given temperature
-    void refrigerate(float temperature);
+    bool refrigerate();
 };
 
+bool VendingMachine::ItemDispenser::distributeItems(int numberOfItems)
+{
+    bool isDispensed = false;
+    
+    if (inventory >= numberOfItems)
+    {
+        inventory -= numberOfItems;
+        isDispensed = true;
+    }
+    else
+    {
+        disable("Empty");
+    }    
+    
+    return isDispensed;
+}
+
+void VendingMachine::ItemDispenser::stockUp(int itemsAdded)
+{
+    inventory += itemsAdded;
+}
+
+void VendingMachine::ItemDispenser::disable(std::string cause)
+{
+    if (cause == "Empty") 
+    {
+        name = "";
+        flavour = "";
+    }
+}
+
+float VendingMachine::chargeCustomerEuros(ItemDispenser item, int numberOfItems)
+{
+    float priceCharged = 0.0f;
+    for (int i = 0; i < numberOfItems; i++)
+    {
+        priceCharged += item.priceEuros;        
+    }
+    return priceCharged;
+}
+
+bool VendingMachine::dispenseItem(ItemDispenser itemDispenser, int numberOfItems)
+{
+    bool isDispensed;
+    
+    itemDispenser.distributeItems(numberOfItems);
+    isDispensed = itemDispenser.isDisabled;
+
+    return isDispensed;
+}
```

Comment:
Don't waste time writing temporary variables like this. 
just write `return itemDispenser.isDisabled;`
- [ ] revise